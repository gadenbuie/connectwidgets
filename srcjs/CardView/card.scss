.rsccard {
  display: flex;
  flex-direction: column;

  &__card {
    border-radius: 3px;
    margin-bottom: 2em;
    position: relative;
    width: 100%;

    & .card-body {
      align-items: flex-start;
      display: flex;
      padding: 0;
    }
  }

  &__img {
    border-radius: 3px;
    box-shadow: 0 0 3px 1px rgba(0,0,0, 0.2);
    flex: 1;
    margin: 16px;
    overflow: hidden;
    position: relative;
  }

  &__meta {
    flex: 3;
    margin: 16px;
    margin-left: 0;

    /*
    &-author {
      default styles on theming files
    }
    */

    &-title {
      margin: 0;
      line-height: 1.72em;
    }

    &-description {
      line-height: 1.62em;
    }
  }
}
