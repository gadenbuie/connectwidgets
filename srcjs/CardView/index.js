import { reactWidget } from 'reactR';
import CardView from './CardView';

reactWidget('rsc_card', 'output', { CardView });
