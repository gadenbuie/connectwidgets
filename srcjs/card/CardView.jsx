/**
 * rsccard main component
 * React provided at window via reactR
 */

import PropTypes from 'prop-types';
import ContentImage from '@/image';
import { contentImgSrc } from '@/utils';
import './card.scss';

function Card({ data }) {
  const contentObj = HTMLWidgets.dataframeToD3(data)[0];
  const imgUrl = contentImgSrc(contentObj.url, contentObj.guid);

  return (
    <div className="rsccard">
      <div
        className='rsccard__img'
      >
        <ContentImage
          imageUrl={imgUrl}
          contentUrl={contentObj.url}
          contentType={contentObj.app_mode}
        />
      </div>
      <div className="rsccard__meta">
        <span>{contentObj.owner_username}</span> â€¢ <time>{new Date(contentObj.updated_time).toDateString()}</time>
        <h2 className="rsccard__meta-title">
          {contentObj.title || contentObj.name}
        </h2>
        <p className="rsccard__meta-description">
          {contentObj.description}
        </p>
        <a
          className="rsccard__meta-link"
          href={contentObj.url}
          target="_blank"
          rel="noreferrer"
        >Open Content</a>
      </div>
    </div>
  );
}

Card.propTypes = {
  data: PropTypes.object,
}

export default Card;
