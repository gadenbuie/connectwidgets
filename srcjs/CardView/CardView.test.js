import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CardView from './CardView';

// Object format frame-like that comes from R
const dataSingleItem = {
  url: ['https://connect.tester.net/content/991f16c5-dc7e-4403-89d0-c54d10968460/'],
  guid: ['991f16c5-dc7e-4403-89d0-c54d10968460'],
  owner_username: ['Porter_Terry'],
  title: ['methodical synergize'],
  name: ['methodical-synergize'],
  description: ['Lorem ipsum dolor sit amet'],
  updated_time: ['Fri Oct 30 2020'],
}

// Object format frame-like that comes from R
const dataMultipleItems = {
  url: [
    'https://connect.tester.net/content/991f16c5-dc7e-4403-89d0-c54d10968460/',
    'https://connect.tester.net/content/c2bd6807-cb2f-4683-865b-107cf37eddf6/',
    'https://connect.tester.net/content/ea731b93-458f-4552-becf-b2e0ce552e3d/',
  ],
  guid: [
    '991f16c5-dc7e-4403-89d0-c54d10968460',
    'c2bd6807-cb2f-4683-865b-107cf37eddf6',
    'ea731b93-458f-4552-becf-b2e0ce552e3d',
  ],
  owner_username: [
    'Porter_Terry',
    'Navarro.BigBand',
    'Bob.C.Martin',
  ],
  title: [
    'methodical synergize',
    'incredible peanuts',
    'clean coder',
  ],
  name: [
    'methodical-synergize',
    'incredible peanuts',
    'clean coder',
  ],
  description: [
    'Lorem ipsum dolor sit amet',
    'Maecenas more than ipsum and three dogs',
    'a book on the shelve',
  ],
  updated_time: [
    'Fri Oct 30 2020',
    'Wed Sep 16 2020',
    'Mon Sep 21 2020',
  ],
}

describe('<CardView />', () => {

  // Snapshot testing is enough for this component
  // as it doesn't bring anything other than presentation

  it('single item render properly', () => {
    const wrapper = shallow(<CardView data={dataSingleItem} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.rsccard__card').length).toBe(1);
  });

  it('mutliple items render properly', () => {
    const wrapper = shallow(<CardView data={dataMultipleItems} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('.rsccard__card').length).toBe(3);
  });
});
