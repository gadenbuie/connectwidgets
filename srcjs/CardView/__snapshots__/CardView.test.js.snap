// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardView /> renders properly 1`] = `
<div
  className="rsccard"
>
  <div
    className="rsccard__img"
  >
    <ContentImage
      contentUrl="https://connect.tester.net/content/991f16c5-dc7e-4403-89d0-c54d10968460/"
      imageUrl="https://connect.tester.net/__api__/applications/991f16c5-dc7e-4403-89d0-c54d10968460/image"
    />
  </div>
  <div
    className="rsccard__meta"
  >
    <span>
      Porter_Terry
    </span>
     â€¢ 
    <time>
      Fri Oct 30 2020
    </time>
    <h2
      className="rsccard__meta-title"
    >
      methodical synergize
    </h2>
    <p
      className="rsccard__meta-description"
    >
      Lorem ipsum dolor sit amet
    </p>
    <a
      className="rsccard__meta-link"
      href="https://connect.tester.net/content/991f16c5-dc7e-4403-89d0-c54d10968460/"
      rel="noreferrer"
      target="_blank"
    >
      Open Content
    </a>
  </div>
</div>
`;
